File: backend/src/reader/dto/reader.dto.ts
import { UUID } from "crypto";

export interface ReaderDto {
    id: UUID,
    firstName: string,
    lastName: string,
    email: string,
    phone: string,
    address: string,
    createdAt: Date,
    updatedAt: Date,
    deletedAt: Date,
    deleted: boolean,
    deletedBy: string,
    updatedBy: string,
    createdBy: string,
}

File: spring-backend/src/main/java/library/app/backend/dto/ReaderDto.java
package library.app.backend.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.io.Serializable;
import java.time.Instant;

/**
 * DTO for {@link library.app.backend.models.Reader}
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ReaderDto(String id, String firstName, String lastName, String email, String phone, String address,
                        Instant createdAt) implements Serializable {
}